import/src/test/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreServiceTest.java

```diff
@@ -287,7 +306,80 @@ public void testCStoreService_transcodeToJpeg2k() throws Exception {
         UID.SecondaryCaptureImageStorage,
         "1.2.276.0.7230010.3.1.4.1784944219.230771.1519337370.699151",
         TestUtils.TEST_IMG_FILE,
-        UID.JPEG2000);
+        UID.JPEG2000,
+        null);
+  }
+
+  @Test
+  public void testBackupUploadService_ExceptionOnWriteToBackup_bytesToStowRsNotAvailable() throws Exception {
+    IBackupUploader mockBackupUploader = mock(IBackupUploader.class);
+    DelayCalculator mockDelayCalculator = mock(DelayCalculator.class);
+    MockStowClient spyStowClient = spy(new MockStowClient(
+        false,
+        HttpStatusCodes.STATUS_CODE_OK));
+
+    doThrow(new IBackupUploader.BackupException("errMsg")).when(mockBackupUploader).doWriteBackup(any(InputStream.class), anyString());
+    doNothing().when(spyStowClient).stowRs(any(InputStream.class));
+
+    BackupUploadService backupUploadService = new BackupUploadService(mockBackupUploader, mockDelayCalculator);
+
+    basicCStoreServiceTest(
+        Status.ProcessingFailure,
+        backupUploadService,
+        spyStowClient);
+
+    verify(mockBackupUploader, never()).doReadBackup(anyString());
+    verify(spyStowClient).stowRs(argThat(is -> equalInputStreamAvailableBytes(is, 0)));
```

Comment:
I'm concerned that equalInputStreamAvailableBytes may lead to flaky tests. Can we just verify that stowRS wasn't called here?

Reply:
test updated. stowRS would be called in any case. (Callable with stowRS Processor ends before backup processor throws exception in case of fail)